libp2p/protocols/kademlia/xordistance.nim

```diff
@@ -2,8 +2,27 @@ import ./consts
 import ./keys
 import nimcrypto/sha2
 import ../../peerid
+import results
 
 type XorDistance* = array[IdLength, byte]
+type XorDHasher* = proc(input: seq[byte]): array[IdLength, byte] {.
+  raises: [], nimcall, noSideEffect, gcsafe
+.}
+
+proc defaultHasher(
+    input: seq[byte]
+): array[IdLength, byte] {.raises: [], nimcall, noSideEffect, gcsafe.} =
+  return sha256.digest(input).data
+
+# useful for testing purposes
+proc noOpHasher*(
+    input: seq[byte]
+): array[IdLength, byte] {.raises: [], nimcall, noSideEffect, gcsafe.} =
+  var data: array[IdLength, byte]
+  for i in 0 ..< min(IdLength, input.len):
+    # memcopy instead?
```

Comment:
yes ☺️

Reply:
ahhh, yes. remember putting in that comment. got lazy about the follow up. I'll change to use memcpy, then run the merge. cheers.